lib/highline/terminal/unix_stty.rb

```diff
@@ -0,0 +1,91 @@
+class HighLine
+  class Terminal::UnixStty < Terminal
+
+    # A Unix savvy method using stty to fetch the console columns, and rows.
+    # ... stty does not work in JRuby
+    def terminal_size
+      begin
+        require "io/console"
+        winsize = IO.console.winsize.reverse rescue nil
+        return winsize if winsize
+      rescue LoadError
+      end
+
+      if /solaris/ =~ RUBY_PLATFORM and
+        `stty` =~ /\brows = (\d+).*\bcolumns = (\d+)/
+        [$2, $1].map { |c| x.to_i }
+      elsif `stty size` =~ /^(\d+)\s(\d+)$/
+        [$2.to_i, $1.to_i]
+      else
+        [ 80, 24 ]
+      end
+    end
+
+                              # *WARNING*:  This requires the external "stty" program!
+    CHARACTER_MODE = "unix_stty"   # For Debugging purposes only.
+
+    def raw_no_echo_mode
+      @state = `stty -g`
+      system "stty raw -echo -icanon isig"
```

Comment:
More clearly about SystemExtensions::Stty...

The only file that is currently working well at lib/highline/terminal (https://github.com/abinoam/highline/tree/feat_refac_ask/lib/highline/terminal)

is unix_stty.rb (https://github.com/abinoam/highline/blob/feat_refac_ask/lib/highline/terminal/unix_stty.rb)

Look that it has 
```ruby
class HighLine
  class Terminal
    class UnixStty < Terminal
```
The other files are the remainings of the pieces of code that used to work for different "terminals".

They are still like 
```ruby 
class HighLine
  module SystemExtensions
    module JRuby
```

As soon as I progress and have a more stable code for UnixStty I'll put those code remains back to life!!!